plugins/module_utils/proxmox.py

```diff
@@ -170,13 +189,28 @@ def get_pool(self, poolid):
         except Exception as e:
             self.module.fail_json(msg="Unable to retrieve pool %s information: %s" % (poolid, e))
 
+    def get_storage(self, storage):
+        try:
+            storage = self.proxmox_api.storage.get(storage)
+        except Exception:
+            self.module.fail_json(msg="Storage '%s' does not exist" % storage)
+        return ProxmoxStorage(storage)
+
     def get_storages(self, type):
         """Retrieve storages information
 
         :param type: str, optional - type of storages
         :return: list of dicts - array of storages
         """
         try:
-            return self.proxmox_api.storage.get(type=type)
+            storages = self.proxmox_api.storage.get(type=type)
+            storages = [ProxmoxStorage(storage) for storage in storages]
+            return storages
```

Comment:
Note that this is a breaking change since the `proxmox` module utils is a public API of the collection.

I would rather add a new function `get_storages_as_objects` that calls `get_storages` that returns these objects.